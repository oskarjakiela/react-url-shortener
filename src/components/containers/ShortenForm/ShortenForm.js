import React from 'react';

import './ShortenForm.css';
import { Button, Input } from 'components/ui';


const displayName = 'Shorty-ShortenForm';

const ShortenForm = () => (
  <form className={displayName}>
    <Input
      className={`${displayName}__input`}
      placeholder="Paste the link you want to shorten here"
    />

    <Button
      className={`${displayName}__button`}
      disabled
    >
      Shorten this link
    </Button>
  </form>
);

ShortenForm.displayName = displayName;

export default ShortenForm;
