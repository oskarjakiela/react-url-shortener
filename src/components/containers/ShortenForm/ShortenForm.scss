.Shorty-ShortenForm {
  display: flex;
  margin: 0 -.5rem;
  justify-content: space-between;

  > * {
    margin: 0 .5rem;
  }
}

.Shorty-ShortenForm__input {
  flex: 1;
}

.Shorty-ShortenForm__button {
}
