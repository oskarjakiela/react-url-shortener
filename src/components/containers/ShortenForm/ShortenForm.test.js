import React from 'react';
import renderer from 'react-test-renderer';

import ShortenForm from './ShortenForm';


it('<ShortenForm /> renders correctly', () => {
  const tree = renderer
    .create(<ShortenForm />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
