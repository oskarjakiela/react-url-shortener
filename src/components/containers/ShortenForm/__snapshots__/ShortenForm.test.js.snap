// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShortenForm /> renders correctly 1`] = `
<form
  className="Shorty-ShortenForm"
>
  <input
    className="Shorty-Input Shorty-ShortenForm'__input"
    placeholder="Paste the link you want to shorten here"
  />
  <button
    className="Shorty-Button Shorty-ShortenForm__button Shorty-Button--disabled"
    disabled={true}
  >
    Shorten this link
  </button>
</form>
`;
